tests/boardfarm_plugins/README.md

```diff
@@ -0,0 +1,182 @@
+                "role": "agent",
+                "delay": 7,
+                "conn_cmd": "",
+                "radios": [
+                    {
+                        "type": "radio_dummy",
+                        "iface": "wlan0"
+                    },
+                    {
+                        "type": "radio_dummy",
+                        "iface": "wlan2"
+                    }
+                ]
+            }
+        ]
+    }
+}
+
+```
+Mandatory fields for parent device are:
+* `name` is any unique name at all
+* `board_type` is a name of existing device class
+* `devices` is a list of devices that will be added, must have at least one device.
+Mandatory fields for child devices(those are listed in **devices** section of parent one):
+* `name` is any unique name at all
+* `type` is a name of existing device class
+When running boardfarm we can specify that custom config with in following manner:
+```
+-c /path/to/custom_config.json
+```
+
+### Devices
+There're bunch of devices already exist in `<boardfarm-repo-root>/boardfarm/devices`,
+but if we created `custom_config.json` which uses previously unknown device,
+then we must implement its class in plugin dir `boardfarm_*/devices/<device-name>.py`.
+As template for the device we can use `<boardfarm-repo-root>/devices/base.py`.
+
+### Test suites
+Custom test suite should be present in root of boardfarm plugin dir and its name must
+match `testsuites*.cfg`.
+Suite names are specified in brackets [], test names must match test classes names
+defined in the files at the "boardfarm_*/tests/" directory.
+Sample contents:
+```
+[suite-name]
+TestCaseName0
+
+[another-suite-name]
+TestCaseName0
+TestCaseName1
+TestCaseName2
+```
+
+### Test file contents:
+Each test must be presented as class, for test file template please refer to:
+`boardfarm/tests/bft_base_test.py`
+Above you can see mandatory class members:
+```
+class SampleTest(BftBaseTest):
+    def runTest(self):
+        '''
+        Main test logic must be implemented here.
+        '''
+        print('{}: Running...'.format(self.__class__.__name__))
+
+    def skipTest(self):
+        '''
+        Method invoked if DUT's consoles don't respond,
+        generally considering device as 'dead'.
+        Here we can provide info about possible reasons and suggestions.
+        '''
+        print('{}: Skipping...'.format(self.__class__.__name__))
+
+    def recover(self):
+        '''
+        Method invoked if test fails with an unexpected exception type.
+        According to boardfarm sources, it is considered as deprecated,
+        but it is suggested to leave it as a stub to prevent possible faults.
+        '''
+        print('{}: Recovering...'.format(self.__class__.__name__)
+
+    @classmethod
+    def teardown_class(cls):
+        '''
+        Teardown method, optional for boardfarm tests.
+        Its 'strange' declaration comparing to other test class methods
+        caused only to boardfarm workflow
```

Comment:
What does this mean?

Are you referring to the `@classmethod` bit? There's nothing strange about that, it's a decorator. [Search the web for some explanation](https://www.tutorialsteacher.com/python/classmethod-decorator)

Reply:
> What does this mean?
> 
> Are you referring to the `@classmethod` bit? There's nothing strange about that, it's a decorator. [Search the web for some explanation](https://www.tutorialsteacher.com/python/classmethod-decorator)

I considered `teardown_class` function strange because I don't understand why it was implemented as 'static' one.
Anyway, It seems like that comment is pointless and I'll remove it leaving only function description.